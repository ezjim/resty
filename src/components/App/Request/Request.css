.Request{
  input {
      display: block;
  }

  label {
      padding: 0.2rem 0.4rem;
      margin: 0.5rem;
      cursor: pointer;
  }

  input[type='radio'] {
      display: none;
  }
  /* if we have a input of type radio that is checked and is next to a label then backgorund color red */
  input[type='radio']:checked + label
   {
      background-color: #eee;
  }

  textarea {
      display: block;
      margin: 1rem;
  }

  button {
      display: block;
  }
}
