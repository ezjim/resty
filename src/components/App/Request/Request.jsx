import React from 'react';
import PropTypes from 'prop-types';
import styles from './Request.css';

const Request = ({ url, onUrlChange, body, onBodyChange, onMethodChange, onSubmit }) => (
  <form className={styles.Request} onSubmit={onSubmit}>
    <input type='text' value={url} onChange={onUrlChange} />

    <input id='GET' type='radio' name='method' value='GET' onChange={onMethodChange}></input>
    <label htmlFor='GET'>GET</label>

    <input id='POST' type='radio' name='method' value='POST' onChange={onMethodChange}></input>
    <label htmlFor='POST'>POST</label>

    <input id='PUT' type='radio' name='method' value='PUT' onChange={onMethodChange}></input>
    <label htmlFor='PUT'>PUT</label>

    <input id='PATCH' type='radio' name='method' value='PATCH' onChange={onMethodChange}></input>
    <label htmlFor='PATCH'>PATCH</label>

    <input id='DELETE' type='radio' name='method' value='DELETE' onChange={onMethodChange}></input>
    <label htmlFor='DELETE'>DELETE</label>

    <textarea value={body} onChange={onBodyChange}> </textarea>
    <button>Go!</button>
  </form>
);

Request.propTypes = {
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  body: PropTypes.object.isRequired,
  onUrlChange: PropTypes.func.isRequired,
  onMethodChange: PropTypes.func.isRequired,
  onBodyChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired

};

export default Request;
