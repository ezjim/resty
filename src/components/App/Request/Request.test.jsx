import React from 'react';
import { shallow } from 'enzyme';
import Request from './Request';

describe('Request', () => {
  it('matches Request jsx', () => {
    const wrapper = shallow(<Request
    
      url='https://hey-arnold-api.herokuapp.com/api/v1/characters'
      method='POST'
      body='json body'
      onUrlChange={()=>{}}
      onMethodChange={()=>{}}
      onBodyChange={()=>{}}
      onSubmit={()=>{}}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
