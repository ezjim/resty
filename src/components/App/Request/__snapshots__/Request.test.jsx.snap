// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request matches Request jsx 1`] = `
<form
  className="Request"
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value="https://hey-arnold-api.herokuapp.com/api/v1/characters"
  />
  <input
    id="GET"
    name="method"
    onChange={[Function]}
    type="radio"
    value="GET"
  />
  <label
    htmlFor="GET"
  >
    GET
  </label>
  <input
    id="POST"
    name="method"
    onChange={[Function]}
    type="radio"
    value="POST"
  />
  <label
    htmlFor="POST"
  >
    POST
  </label>
  <input
    id="PUT"
    name="method"
    onChange={[Function]}
    type="radio"
    value="PUT"
  />
  <label
    htmlFor="PUT"
  >
    PUT
  </label>
  <input
    id="PATCH"
    name="method"
    onChange={[Function]}
    type="radio"
    value="PATCH"
  />
  <label
    htmlFor="PATCH"
  >
    PATCH
  </label>
  <input
    id="DELETE"
    name="method"
    onChange={[Function]}
    type="radio"
    value="DELETE"
  />
  <label
    htmlFor="DELETE"
  >
    DELETE
  </label>
  <textarea
    onChange={[Function]}
    value="json body"
  >
     
  </textarea>
  <button>
    Go!
  </button>
</form>
`;
